### PR TITLE
fix puppetagent inputs plugin to support strings for config variable for environments

### PR DESCRIPTION
**Pull request by frankxoom**
### Required for all PRs:
- [x] CHANGELOG.md updated (we recommend not updating this until the PR has been approved by a maintainer)
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [x] README.md updated (if adding a new plugin)

### CODE REVIEW

**Comment by sparrc** [comment]
can you explain what this change is about? why is it necessary? has a puppet agent field changed in a newer version?

**Comment by sparrc** [comment]
@frankxoom nevermind, Sean pointed to the zendesk ticket referencing this.

I agree that this field should be a string, but unfortunately this would be a breaking change for anyone currently using the plugin. They will receive an error saying that the `version_config` field already exists as an int, and can't be written as a string.

for this reason, could you also change the name of the field to `version_config_string`? that way the `version_config` int field will simply get dropped.

You'll also need to get the unit tests passing before I can merge.

**Comment by frankxoom** [comment]
@sparrc I made the changes you suggested, and the unit tests have passed.   Let me know if there is something else you would like changed.

Thanks for your assistance.

--Frank Stutz

File: plugins/inputs/puppetagent/puppetagent.go [hunk 1]
```diff
@@ -68,8 +68,8 @@ type time struct {
 }
 
 type version struct {
-	Config string `yaml:"config"`
-	Puppet string `yaml:"puppet"`
+	Config_string string `yaml:"config"`
```
Comment (sparrc) [inline plugins/inputs/puppetagent/puppetagent.go]: this should be ConfigString

File: plugins/inputs/puppetagent/puppetagent.go [hunk 2]
```diff
@@ -68,8 +68,8 @@ type time struct {
 }
 
 type version struct {
-	Config int64  `yaml:"config"`
-	Puppet string `yaml:"puppet"`
+	Config_string string `yaml:"config"`
```
Comment (sparrc) [inline plugins/inputs/puppetagent/puppetagent.go]: this should be ConfigString

**Comment by sparrc** [comment]
thanks @frankxoom update the casing, sign the CLA, and update the changelog and I can merge this

**Comment by frankxoom** [comment]
@sparrc 

Okay, I updated casing, changelog (I put in 1.2 since you put in that milestone), and signed CLA.  Thanks for your help.

**Pull request closed without merging**